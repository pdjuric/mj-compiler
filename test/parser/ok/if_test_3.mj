program if_test

int testCaseNo;
{
    bool T(int t) {
        print(t, 2);
        return true;
    }

    bool F(int t) {
        print(t, 2);
        return false;
    }

    void inThen(int i) {
        print('t',2);
        print('h',1);
        print('e',1);
        print('n',1);
        print(i,2);
    }

    void inElse(int i) {
        print('e',2);
        print('l',1);
        print('s',1);
        print('e',1);
        print(i,2);
    }

    void next() {
        print(eol);
        print('t',1);
        print('e',1);
        print('s',1);
        print('t',1);
        print(testCaseNo, 4);
        print(':', 1);
        print(' ', 1);
        testCaseNo++;
    }

    void main() {
        testCaseNo = 0;

		next();        if (T(0)) if (T(1)) inThen(1);
		next();        if (T(0)) if (F(1)) inThen(1);
		next();        if (F(0)) if (T(1)) inThen(1);
		next();        if (F(0)) if (F(1)) inThen(1);

		next();        if (T(0)) if (T(1)) inThen(1); else inElse(1);
		next();        if (T(0)) if (F(1)) inThen(1); else inElse(1);
		next();        if (F(0)) if (T(1)) inThen(1); else inElse(1);
		next();        if (F(0)) if (F(1)) inThen(1); else inElse(1);

		next();        if (T(0)) if (T(1)) inThen(1); else inElse(1); else inElse(0);
		next();        if (T(0)) if (F(1)) inThen(1); else inElse(1); else inElse(0);
		next();        if (F(0)) if (T(1)) inThen(1); else inElse(1); else inElse(0);
		next();        if (F(0)) if (F(1)) inThen(1); else inElse(1); else inElse(0);

		next();        if (T(0)) inThen(0); else if (T(1)) inThen(1);
		next();        if (T(0)) inThen(0); else if (F(1)) inThen(1);
		next();        if (F(0)) inThen(0); else if (T(1)) inThen(1);
		next();        if (F(0)) inThen(0); else if (F(1)) inThen(1);

		next();        if (T(0)) inThen(0); else if (T(1)) inThen(1); else inElse(1);
		next();        if (T(0)) inThen(0); else if (F(1)) inThen(1); else inElse(1);
		next();        if (F(0)) inThen(0); else if (T(1)) inThen(1); else inElse(1);
		next();        if (F(0)) inThen(0); else if (F(1)) inThen(1); else inElse(1);

		next();        if (T(0)) if (T(1)) if (T(2)) inThen(2); else inElse(2); else inElse(1);
		next();        if (T(0)) if (T(1)) if (F(2)) inThen(2); else inElse(2); else inElse(1);
		next();        if (T(0)) if (F(1)) if (T(2)) inThen(2); else inElse(2); else inElse(1);
		next();        if (T(0)) if (F(1)) if (F(2)) inThen(2); else inElse(2); else inElse(1);
		next();        if (F(0)) if (T(1)) if (T(2)) inThen(2); else inElse(2); else inElse(1);
		next();        if (F(0)) if (T(1)) if (F(2)) inThen(2); else inElse(2); else inElse(1);
		next();        if (F(0)) if (F(1)) if (T(2)) inThen(2); else inElse(2); else inElse(1);
		next();        if (F(0)) if (F(1)) if (F(2)) inThen(2); else inElse(2); else inElse(1);


		next();        if (T(0)) if (T(1)) if (T(2)) inThen(2); else if (T(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (T(0)) if (T(1)) if (F(2)) inThen(2); else if (T(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (T(0)) if (F(1)) if (T(2)) inThen(2); else if (T(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (T(0)) if (F(1)) if (F(2)) inThen(2); else if (T(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (F(0)) if (T(1)) if (T(2)) inThen(2); else if (T(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (F(0)) if (T(1)) if (F(2)) inThen(2); else if (T(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (F(0)) if (F(1)) if (T(2)) inThen(2); else if (T(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (F(0)) if (F(1)) if (F(2)) inThen(2); else if (T(3)) inThen(3); else inElse(3); else inElse(1);

		next();        if (T(0)) if (T(1)) if (T(2)) inThen(2); else if (F(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (T(0)) if (T(1)) if (F(2)) inThen(2); else if (F(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (T(0)) if (F(1)) if (T(2)) inThen(2); else if (F(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (T(0)) if (F(1)) if (F(2)) inThen(2); else if (F(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (F(0)) if (T(1)) if (T(2)) inThen(2); else if (F(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (F(0)) if (T(1)) if (F(2)) inThen(2); else if (F(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (F(0)) if (F(1)) if (T(2)) inThen(2); else if (F(3)) inThen(3); else inElse(3); else inElse(1);
		next();        if (F(0)) if (F(1)) if (F(2)) inThen(2); else if (F(3)) inThen(3); else inElse(3); else inElse(1);


    }

}