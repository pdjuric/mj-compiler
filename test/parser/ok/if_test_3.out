
test   0:  0 1 then 1
test   1:  0 1
test   2:  0
test   3:  0
test   4:  0 1 then 1
test   5:  0 1 else 1
test   6:  0
test   7:  0
test   8:  0 1 then 1
test   9:  0 1 else 1
test  10:  0 else 0
test  11:  0 else 0
test  12:  0 then 0
test  13:  0 then 0
test  14:  0 1 then 1
test  15:  0 1
test  16:  0 then 0
test  17:  0 then 0
test  18:  0 1 then 1
test  19:  0 1 else 1
test  20:  0 1 2 then 2
test  21:  0 1 2 else 2
test  22:  0 1 else 1
test  23:  0 1 else 1
test  24:  0
test  25:  0
test  26:  0
test  27:  0
test  28:  0 1 2 then 2
test  29:  0 1 2 3 then 3
test  30:  0 1 else 1
test  31:  0 1 else 1
test  32:  0
test  33:  0
test  34:  0
test  35:  0
test  36:  0 1 2 then 2
test  37:  0 1 2 3 else 3
test  38:  0 1 else 1
test  39:  0 1 else 1
test  40:  0
test  41:  0
test  42:  0
test  43:  0